### PR TITLE
feat(lombok-devfile): Add lombok devfile

### PR DESCRIPTION
**Pull request by sunix**
### What does this PR do?
Provide a lombok stack with.
- a sample project
- the Java plugin properly configured

### Screenshot/screencast of this PR
image

Selection_268

### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/17483

### How to test this PR?
I have pushed a devfile registry with these changes in surge.sh. To test it, update the CheCluster CR with:
```
    devfileRegistryUrl: https://sunix-che-devfile-registry-lombok-qpd6g.surge.sh
    externalDevfileRegistry: true
```

Wait for the dashboard to reload.

Some problems you may have while testing:
- you may need to disable CORS: `chromium-browser --disable-web-security --user-data-dir=/tmp/chromium-tmp-user-data-cors`
- the devfile requires https://github.com/eclipse/che-plugin-registry/pull/871 So if your are not using nightly nor 7.28, update the plugin with the registry-url in the devfile:
   ```yaml
     - id: redhat/java/latest
       preferences:
         java.jdt.ls.vmargs: '-javaagent:/lombok.jar'
         java.server.launchMode: Standard
       registryUrl: 'https://sunix-plugin-registry-lombok-location-esc6v.surge.sh/v3'

   ```

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [x] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [x] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [x] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [ ] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [x] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [x] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [ ] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### CODE REVIEW

File: devfiles/lombok/devfile.yaml [hunk 1]
```diff
@@ -0,0 +1,58 @@
+---
+apiVersion: 1.0.0
+metadata:
+  generateName: java-lombok-
+
+projects:
+
+  - name: lombok-project-example
+    source:
+      location: 'https://github.com/chuucks/LOMBOK-PROJECT-EXAMPLE'
```
Comment (ericwill) [inline devfiles/lombok/devfile.yaml:10]: We may want to fork this to che-samples: https://github.com/che-samples

**Comment by sunix** [comment]
the icon looks too big

Selection_267

**Comment by sunix** [comment]
I have fixed the image size

**Review by ericwill — APPROVED** [review]
LGTM :+1:

**Pull request merged**